torch/distributed/fsdp/flat_param.py

```diff
@@ -617,6 +673,8 @@ def pre_unshard(self) -> bool:
         matches the dtype of the expected unsharded parameter.
         """
         ret = False
+        if self._use_orig_params:
+            ret = ret or self._writeback_orig_params()
```

Comment:
ret seems always `False` here, so this is just `ret = self._writeback_orig_params`?

Reply:
Yes. I included `ret = ret or ...` in case we ever add another call before `_writeback_orig_params()` to avoid forgetting to adjust L677. It is fine either way.